kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/sink/PipelineDataSourceSink.java

```diff
@@ -105,7 +105,10 @@ private PipelineJobProgressUpdatedParameter flush(final DataSource dataSource, f
             flushInternal(dataSource, each.getBatchDeleteDataRecords());
             flushInternal(dataSource, each.getBatchInsertDataRecords());
             flushInternal(dataSource, each.getBatchUpdateDataRecords());
-            sequentialFlush(dataSource, each.getNonBatchRecords());
+            if (each.getNonBatchRecords().isEmpty()) {
+                continue;
+            }
+            tryFlush(dataSource, each.getNonBatchRecords(), false);
```

Comment:
Could we use for each of `each.getNonBatchRecords()` and invoke `doFlush`? Then `doFlush` could be simplified.